flytekit/common/workflow.py

```diff
@@ -192,12 +200,16 @@ def promote_from_model(cls, base_model):
         for v in _six.itervalues(node_map):
             v.upstream_nodes[:] = [node_map[k] for k in v.upstream_node_ids]
 
+        iface = _interface.TypedInterface.promote_from_model(base_model.interface)
+        inputs = [_Input(name=k, sdk_type=_type_helpers.get_sdk_type_from_literal_type(v.type), required=True) for k, v
+                  in _six.iteritems(iface.inputs)]
+        outputs = [Output(name=o.var, value=None, binding_data=o.binding,
```

Comment:
perhaps we can use the above created node map to simply create these as normal `Output(node_map[o.binding.node_id].outputs[o.binding.var], sdk_type=...)` and not alter the constructor (see above comment).